File: src/org/cappuccino_project/ide/intellij/plugin/psi/interfaces/ObjJHasFunctionName.java
package org.cappuccino_project.ide.intellij.plugin.psi.interfaces;

import com.intellij.psi.PsiElement;
import org.jetbrains.annotations.NotNull;

import javax.annotation.Nullable;

public interface ObjJHasFunctionName {
    @NotNull
    String getFunctionNameAsString();

    @Nullable
    ObjJNamedElement getFunctionNameNode();
}

File: src/main/java/cappuccino/ide/intellij/plugin/psi/interfaces/ObjJHasFunctionName.kt
package cappuccino.ide.intellij.plugin.psi.interfaces

import cappuccino.ide.intellij.plugin.universal.psi.ObjJUniversalPsiElement

interface ObjJHasFunctionName: ObjJUniversalPsiElement {
    val functionNameString: String

    val functionNameNode: ObjJNamedElement?
}
